sessionctx/variable/sysvar.go

```diff
@@ -492,12 +492,40 @@ func UnregisterSysVar(name string) {
 	sysVarsLock.Unlock()
 }
 
+// copySv deep copies the sysvar struct to avoid a race
+func copySv(src *SysVar) (dest *SysVar) {
+	return &SysVar{
```

Comment:
I think you might be able to do a trick like `dst := *src, return &dst`. But neither your or mine is a complete deep copy, slice is essentially a reference. Should ensure that slice is not modified but replaced.

Reply:
We can add `Clone()` method for `*SysVar` and put it next to the struct definition.

Reply:
Thanks for the suggestions, I will change it!

I swear I tried `dst := *src, return &dst` originally :-) It's much cleaner.